vivid-server/src/modules/friends/friends.service.ts

```diff
@@ -12,62 +14,91 @@ export class FriendsService {
     private friendsRepository: Repository<FriendsEntity>,
   ) {}
 
+  // Admin only, find all FriendsEntity's
   async findAll(): Promise<FriendsEntity[]> {
     return await this.friendsRepository.find();
   }
 
-  async friendRequest(
-    userId: string,
-    friendId: string,
-  ): Promise<Observable<FriendsEntity>> 
-  {
-    let found = await this.friendsRepository.findOne({
-      where: {
-        id_request: userId,
-        id_accept: friendId,
-      }, // let's not forget about when the friend request was already made in the different direction
-    });
-
-    if (found)
-      return ;
-
-    return from(
-      this.friendsRepository.save({
-        id_request: userId,
-        id_accept: friendId,
-      }),
-    );
+  // Add FriendEntity to database (=> send friend request)
+  async sendFriendRequest(
+    user_1: string,
+    user_2: string,
+    requested_by: string,
+    requested_to: string,
+  ) {
+    return await this.friendsRepository
+      .createQueryBuilder()
+      .insert()
+      .values([
+        {
+          user_1: user_1,
+          user_2: user_2,
+          requested_by: requested_by,
+          requested_to: requested_to,
+        },
+      ])
+      .execute()
+      .catch((error) => {
+        if ((error.code = '23505')) throw new BadRequestException();
+      });
   }
 
-  async findRequests(
-    userId: string)
-    :Promise<FriendsEntity[]>
-  {
-    let requests =  await this.friendsRepository.find({
-      where: {
-        id_accept: userId,
-        accepted: false,
-      },
-    });
-    console.log(requests);
-    return requests;
+  // Find user's pending friend requests (FriendsEntity's that aren't accepted)
+  async findAllFriendRequests(userId: string): Promise<FriendsEntity[]> {
+    return await this.friendsRepository
+      .createQueryBuilder()
+      .select()
+      .where('requested_to = :r', { r: userId })
+      .andWhere('accepted = :a', { a: false })
+      .execute();
   }
 
-  async acceptRequest(
-    userId: string,
-    friendId: string,
-  )
-  {
-    let found = await this.friendsRepository.findOne({
-      where: {
-        id_request: friendId,
-        id_accept: userId,
-        accepted: false
-      },
-    });
-    if (!found)
-      return ;
-    this.friendsRepository.update(found.id, {accepted: true});
+  // find all friends of the user
+  async getFriendList(userId: string) {
+    return await getConnection()
+      .createQueryBuilder()
+      .select()
+      .from((el) => {
+        return el
+          .select(
+            `CASE  WHEN user_1 = '${userId}' THEN user_2 
+                   WHEN user_2 = '${userId}' THEN user_1 
+            END`,
+            'friends',
+          )
+          .from(FriendsEntity, 'f')
+          .where('accepted = :a', { a: true });
+      }, 'f')
```

Comment:
This is vulnerable to sql injection. use
```js
query.select( `CASE  WHEN user_1 = :u THEN user_2 WHEN user_2 = :u THEN user_1 END`, 'friends').setParameter('u', 123);
.where("user_1 = :u OR user_2 = :u", { u: 123 }).andWhere("accepted = :a", { a: true })
```
instead of this.